lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -191,7 +191,7 @@ defmodule LambdaEthereumConsensus.ForkChoice do
          {:ok, new_store} <-
            signed_block.message.body.attester_slashings
            |> apply_handler(new_store, &Handlers.on_attester_slashing/2) do
-      {:ok, new_store}
+      {:ok, Handlers.prune_checkpoint_states(new_store)}
```

Comment:
shouldn't we do this at the end of an epoch instead of on every block?

Reply:
I did it this way for simplicity. The map is actually _really_ small in size (~3 key-value pairs in the common case), and it doesn't impact performance.